custom/conf/app.example.ini

```diff
@@ -49,8 +49,7 @@ APP_NAME = ; Gitea: Git with a cup of tea
 ;; RUN_USER will automatically detect the current user - but you can set it here change it if you run locally
 RUN_USER = ; git
 ;;
-;; Application run mode, affects performance and debugging: "dev" or "prod", default is "prod"
-;; Mode "dev" makes Gitea easier to develop and debug, values other than "dev" are treated as "prod" which is for production use.
+;; Application run mode: `dev` or `prod`, default is `prod`. Mode `dev` is only meant for Gitea development and must not be used for any other purpose. Values other than `dev` are treated as `prod`.
```

Comment:
If that is being done, then I could hook it to `GITEA_RUN_MODE=dev` like I previously suggested already. It's the safer option and not breaking.

Reply:
Sorry I do not understand how "hook it to GITEA_RUN_MODE=dev" works. It would break many developers who uses GoLand as their IDE like me. Personally I do not like to be forced to use `make watch`, it's slow and buggy.

Reply:
I certainly prefer the env var approach via Makefile over another ini config setting that every dev has to set when setting up their dev instance.

Reply:
The problems of `make watch` IMO:

* I do not want to keep it building, sometimes I changes files frequently, I know when I need to build while when not, the `watch` just wastes my CPU and slows things down
* I do not want to re-build sometimes if a file is changed, sometimes I just need to confirm something by existing process when changing source code.
* It is buggy ....  I have helped the `make watch` users to figure out the bugs like #24845

Reply:
GITEA_RUN_MODE is set here: https://github.com/go-gitea/gitea/blob/8228751c55d6a4263f0fec2932ca16181c09c97d/Makefile#L431, which should already override RUN_MODE from the ini currently.

Reply:
How about keep "mode=dev" as before, and introduce "options for run-mode"?

eg: 

* `RUN_MODE = prod`, as before
* `RUN_MODE = dev`, as before
* `RUN_MODE = dev:assets-port=3001` / `GITEA_RUN_MODE = dev:assets-port=3001` : proxy "/assets" to port 3001

Then this PR is not breaking, and doesn't need the extra GITEA_DEV_FRONTEND_PORT.

Reply:
I guess this could work, I would probably do it in `Makefile` via `GITEA_RUN_MODE` by default but anyone not using `Makefile` needs to configure it themselves.

Also I think I may add a new target instead of `watch` for this in case people still prefer the `watch` mode to only watch.